.github/workflows/codeql-analysis.yml

```diff
@@ -18,9 +18,23 @@ name: CodeQL
 on:
   push:
     branches: [ main ]
+    paths:
+      - '**'
+      - '!**.md'
+      - '!**.txt'
+      - '!images/**'
+      - '!LICENSE'
+      - 'tests/**'
   pull_request:
     # The branches below must be a subset of the branches above
     branches: [ main ]
+    paths:
```

Comment:
this is not good because this job is required and then this will never run or skipped and will be pending in the github status

I think for everything we should use the `dorny/paths-filter`

```
      - uses: dorny/paths-filter@v2
        id: filter
        with:
          filters: |
            terraform:
              - 'tests/**'
```

we will need to check if the syntax `- '!images/**'` works with this

Reply:
Could you explain more the first part: "will never run or skipped" ?

Also, `dorny/paths-filter` seems to handle the syntax `!images/**` in another way:
- Define the filters (paths that can trigger the job or a step in a workflow).
- Use `if` statement to choose if you want to trigger the job or not, based on the existence of the file or not.